File: hw/vargas/u3/hw28PrinciplesAplication/java/ToolsManagement/src/main/java/ec/edu/espe/tools/utils/IToolView.java
package ec.edu.espe.tools.utils;

import ec.edu.espe.tools.controller.ToolController;
import ec.edu.espe.tools.model.Tool;
import java.util.List;

/**
 *
 * @author Cesar Vargas, Paradigm, @ESPE
 */
public interface IToolView {
    String getIdInput();
    String getNameInput();
    double getPriceInput();
    String getMaterialsInput();
    
    void showMessage(String msg);
    void showError(String error);
    void refreshTable(List<Tool> tools);
    void clearFields();
    
    void setController(ToolController controller);
}

File: hw/vargas/u3/hw28PrinciplesAplication/python/ITool_View.py
from abc import ABC, abstractmethod
from typing import List

class ITool_View(ABC):
    @abstractmethod
    def get_form_data(self) -> dict: pass
    
    @abstractmethod
    def show_message(self, title: str, msg: str, is_error: bool = False): pass
    
    @abstractmethod
    def update_list(self, tools: List): pass
    
    @abstractmethod
    def clear_form(self): pass
    
    @abstractmethod
    def set_controller(self, controller): pass